File: Office4U.Articles.Web/src/app/_models/articleParams.ts
import { Article } from './article';

export class ArticleParams {
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'code';

    code: string;
    supplierId: string;
    supplierReference: string;
    name1: string;
    unit: string;
    purchasePriceMin?: number;
    purchasePriceMax?: number;

    constructor() { }
}

File: Office4U.Articles.Common/ArticleParams.cs
namespace Office4U.Articles.Common
{
    public class ArticleParams : ParamsBase
    {
        public string Code { get; set; }
        public string SupplierId { get; set; }
        public string SupplierReference { get; set; }
        public string Name1 { get; set; }
        public string Unit { get; set; }
        public decimal? PurchasePriceMin { get; set; }
        public decimal? PurchasePriceMax { get; set; }
        public string OrderBy { get; set; } = "Code";
    }
}
